### PR TITLE
port ops to JSON (compiler, errors, fetch, files)

### PR DESCRIPTION
**Pull request by bartlomieju**
Based on #2802

- [x] `compiler.rs`
- [x] `errors.rs`
- [x] `fetch.rs`
- [x] `files.rs` (sans, `op_write` and `op_read`)

### CODE REVIEW

**Comment by bartlomieju** [comment]
@ry how should we handle `op_write` and `op_read` that still use flatbuffers? Try to land again #2788 or do them in in JSON as hot fix and revisit later?

**Comment by ry** [comment]
> how should we handle op_write and op_read that still use flatbuffers? 

Just leave them in flatbuffers for now. I'll port them over to dispatch_minimal later.

**Comment by bartlomieju** [comment]
@ry this PR should be finished from my side. Please review, I'll rebase it once you land #2802

**Comment by bartlomieju** [comment]
@ry I rebased this PR on latest master, but it should be landed after #2802 because op ids are incremented from last id used in that PR

File: js/dispatch.ts [hunk 1]
```diff
@@ -22,7 +30,19 @@ export function asyncMsgFromRust(opId: number, ui8: Uint8Array): void {
...
+    case OP_EXEC_PATH:
     case OP_UTIME:
+    case OP_APPLY_SOURCE_MAP:
+    case OP_FORMAT_ERROR:
+    case OP_CACHE:
+    case OP_FETCH_SOURCE_FILE:
+    case OP_OPEN:
+    case OP_CLOSE:
+    case OP_SEEK:
+    case OP_FETCH:
```
Comment (ry) [inline js/dispatch.ts]: Only put async ops in this list - that will induce an error through the default case below if we unexpectedly get an async response for a sync op.

File: js/fetch.ts [hunk 2]
```diff
@@ -320,67 +321,37 @@ export class Response implements domTypes.Response {
...
-  headers: null | domTypes.Headers
-): flatbuffers.Offset {
-  const methodOffset = !method ? 0 : builder.createString(method);
-  let fieldsOffset: flatbuffers.Offset = 0;
-  const urlOffset = builder.createString(url);
+  method: string | null,
+  headers: domTypes.Headers | null,
+  body: ArrayBufferView | undefined
+): Promise<FetchResponse> {
+  const serializedHeaders = [];
```
Comment (ry) [inline js/fetch.ts]: Maybe s/serializedHeaders/headerArray/ ?

maybe you can do 

```
let headerArray = Array.from(headers.entries());
```

File: js/dispatch.ts (hunk 1 above)
Comment (bartlomieju) [reply js/dispatch.ts]: Ah, makes sense! Will update

**Pull request merged**